### PR TITLE
minor buster arm CC nerf & code tweak

### PR DESCRIPTION
**Pull request by LazennG**
# Document the changes in your pull request
nerfs the buster arm a bit but more importantly uses seismic arm to get rid of sleeps
-mop the floor no longer immobilizes victims when they're rammed into things
-grappling people no longer stuns since you'll still be plus as the one initiating the grab
-throwing people into things after grappling immobilizes/knocks down for 1 second instead of 1.5, though people hit by the initial projectile are down for 1.5 still
-buster arm can be attached to others now since the bug that broke shit concerning that is gone too
-got rid of stuff pertaining to old buster actions
-slamming people now leaves damage on the resulting tile

# Why is this good for the game?
 common complaints i hear about the buster arm is sharing the grab>graveyard combo so i'm watering it down a bit so it might be combatable below 30pop

# Testing
in turning the procs into ones that use addtimers i double checked that they worked and function like they used to

# Changelog

:cl:    
tweak: nerfed buster CC effects & turned sleeps into addtimers
/:cl:

### CODE REVIEW

File: code/datums/martial/buster_style.dm [hunk 1]
```diff
@@ -138,16 +132,15 @@
 	if(isstructure(target) || ismachinery(target) || ismecha(target))
 		var/obj/I = target
 		old_density = I.density
-		if(ismecha(I)) // Can pick up mechs
+		if(ismecha(I)) 
 			I.anchored = FALSE
-		if(I.anchored == TRUE) // Cannot pick up anchored structures
-			if(istype(I, /obj/machinery/vending)) // Can pick up vending machines, even if anchored
+		if(I.anchored == TRUE)
```
Comment (Moltijoe) [inline code/datums/martial/buster_style.dm]:
```suggestion
		if(I.anchored) 
```

File: code/datums/martial/buster_style.dm [hunk 2]
```diff
@@ -156,59 +149,51 @@
...
 			I.density = old_density
-		thrown |= I // Mark the item for throwing
+		thrown = I 
 		if(ismecha(I))
 			I.anchored = TRUE
 		return COMSIG_MOB_CANCEL_CLICKON
 	if(isliving(target))
 		var/mob/living/L = target
-		var/obj/structure/bed/grip/F = new(Z, user) // Buckles them to an invisible bed
+		var/obj/structure/bed/grip/F = new(Z, user)
```
Comment (Moltijoe) [inline code/datums/martial/buster_style.dm:167]: Why remove the comment

File: code/datums/martial/buster_style.dm [hunk 3]
```diff
@@ -228,74 +213,84 @@
...
+	soar(user, thrown, D, target_dist, flying)
+
+
+
+
+
+
+/datum/martial_art/buster_style/proc/soar(mob/living/user, atom/movable/center, turf/endzone, distance = 0, list/thrown) //proc for throwing something you picked up with grapple
+	var/slamdam = 7
+	var/objdam = 50
```
Comment (Moltijoe) [inline code/datums/martial/buster_style.dm]: Just make these defines if theyre not going to be changed at any point during use

File: code/datums/martial/buster_style.dm [hunk 4]
```diff
@@ -228,74 +213,84 @@
...
+	if(distance == 0)
+		return
+	var/dir_to_target = get_dir(get_turf(center), endzone) //vars that let the thing be thrown while moving similar to things thrown normally
+	var/turf/next = get_step(get_turf(center), dir_to_target)
+	if(next.density) // crash into a wall and damage everything flying towards it before stopping 
+		for(var/mob/living/S in thrown)
```
Comment (Moltijoe) [inline code/datums/martial/buster_style.dm]: You have two variables named thrown
A list thats specific to the proc
And a variable thats for the entire martial art

File: code/datums/martial/buster_style.dm [hunk 5]
```diff
@@ -307,76 +302,84 @@
 
 /datum/martial_art/buster_style/proc/mop(mob/living/user)
 	var/jumpdistance = 5
-	var/dragdam = 8
-	var/crashdam = 10
```
Comment (Moltijoe) [inline code/datums/martial/buster_style.dm:311]: Defines

**Review by Moltijoe — CHANGES_REQUESTED** [review]
Not done reviewing but this is a start

File: code/datums/martial/buster_style.dm (hunk 2 above)
Comment (LazennG) [reply code/datums/martial/buster_style.dm:167]: was removing a bunch of comments i didnt like and forgot to replace this one oops

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**